moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -161,6 +167,31 @@ struct CollisionResult
     cost_sources.clear();
   }
 
+  /** \brief Throttled log of the first collision pair, if any */
+  void logFirstCollisionPair()
+  {
+    if (!contacts.empty())
+    {
+      ROS_WARN_STREAM_THROTTLE_NAMED(ROS_LOG_THROTTLE_PERIOD, LOGNAME,
+                                     "Objects in collision (among others, possibly): "
+                                         << contacts.begin()->first.first << ", " << contacts.begin()->first.second);
+    }
+  }
+
+  /** \brief Throttled log of all collision pairs */
+  void logAllCollisionPairs()
```

Comment:
I quite liked the distinction between WARN for first and DEBUG for all further collisions.
Do you think it's too much to expect people to understand/use the logger level system for that?

Reply:
i suppose it's not. I'll do what you and Felix both suggested - warn for the first, debug for the rest